src/OnboardingSPA/pages/Steps/Complete/index.js

```diff
@@ -0,0 +1,57 @@
+import { store as nfdOnboardingStore } from '../../../store';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { useNavigate } from 'react-router-dom';
+
+import { StepLoader } from '../../../components/Loaders';
+import { completeFlow } from '../../../utils/api/flow';
+import { StepErrorState } from '../../../components/ErrorState';
+import getContents from './contents';
+import { DesignStateHandler } from '../../../components/StateHandlers';
+
+const StepComplete = () => {
+	const { setIsDrawerSuppressed, setIsHeaderNavigationEnabled } =
+		useDispatch( nfdOnboardingStore );
+
+	const navigate = useNavigate();
+	const [ isError, setIsError ] = useState( false );
+
+	const { nextStep, brandName } = useSelect( ( select ) => {
+		return {
+			nextStep: select( nfdOnboardingStore ).getNextStep(),
+			brandName: select( nfdOnboardingStore ).getNewfoldBrandName(),
+		};
+	}, [] );
+
+	const contents = getContents( brandName );
+
+	useEffect( async () => {
```

Comment:
Maybe take a look at this? Maybe we can factor this out into a function? Just in Case

https://ultimatecourses.com/blog/using-async-await-inside-react-use-effect-hook#:~:text=This%20WORKS%2C%20but%20you%20should%20avoid%20it.%20Why%3F%20Because%20React%E2%80%99s%20useEffect%20hook%20expects%20a%20cleanup%20function%20returned%20from%20it%20which%20is%20called%20when%20the%20component%20unmounts.%20Using%20an%20async%20function%20here%20will%20cause%20a%20bug%20as%20the%20cleanup%20function%20will%20never%20get%20called.%20Yikes!%20So%20what%20do%20we%20do%3F

Reply:
Ohhhh Thanks again! I will keep this in mind and also create stories for wherever we may have missed doing this.